rust-code-analysis-cli/src/formats.rs

```diff
@@ -62,22 +62,20 @@ impl Format {
                 Format::Yaml => ".yml",
             };
 
-            let output_path = output_path.as_ref().unwrap();
+            // Use file path as filename
+            let filename = path.to_str().unwrap();
 
-            let mut file = path.as_path().file_name().unwrap().to_os_string();
-            file.push(format_ext);
+            // Remove ./ ../ .\ ..\ present in a path
+            let re = Regex::new(r"\.\./|\.\.\\|^\./|^\.\\").unwrap();
+            let no_dots_filename = re.replace_all(&filename, "");
```

Comment:
Is there some std lib function to do something like this?

Reply:
I think we could obtain the same result, reducing the code at the same time, using the [components](https://doc.rust-lang.org/std/path/struct.Path.html#method.components) function. I have to try though